dev_docs/cxg.md

```diff
@@ -31,39 +30,44 @@ The `cxg` prefix is used for CXG-specific names.
-* `var`: a TileDB array, of shape (n_var,), containing var annotations, each annotation stored in a separate TileDB array attribute.
-* `X`: a TileDB array, of shape (n_obs, n_var) for dense data, with a single TileDB attribute of numeric type.
-* `Xr`: a TileDB array, of shape (n_obs) for sparse data, with two TileDB attributes of numeric type. The first ("") is expression data, and the second ("var") are the corresponding column indices. Used for row-slicing.
-* `Xc`: a TileDB array, of shape (n_var) for sparse data, with a single TileDB attribute of numeric type. The first ("") is expression data, and the second ("obs") are the corresponding row indices. Used for column-slicing.
-* `X_col_shift`: (optional) TileDB array used in column shift encoding, shape (n_var,), dtype = X.dtype. Single unnamed numeric attribute.  
-* `emb`: a TileDB group, which in turn contains all (zero or more) embeddings.
-* `emb/<EMBEDDING_NAME>`: a TileDB array, with a single anonymous attribute, of numeric type, and shape (n_obs, N>=2).
-* `cxg_group_metadata`: an empty TileDB array, used to store CXG-wide metadata
+The CXG is a TileDB group containing all data and metadata for a single annotated matrix. The following objects MUST be present in a CXG, except where noted as optional:
+
+- `obs`: a TileDB array, of shape (n_obs,), containing obs annotations, each annotation stored in a separate TileDB array attribute. Annotations for categorical metadata may be encoded as integers for faster performance. The integers must correspond to the index location of the corresponding categories in the schema stored in the array's TileDB metadata (`ObsArray.meta['cxg_schema']`).
+- `var`: a TileDB array, of shape (n_var,), containing var annotations, each annotation stored in a separate TileDB array attribute.
+- `X`: a TileDB array, of shape (n_obs, n_var) for dense data, with a single TileDB attribute of numeric type.
+- `Xr`: a TileDB array, of shape (n_obs) for sparse data, with two TileDB attributes of numeric type. The first ("") is expression data, and the second ("var") are the corresponding column indices. Used for row-slicing.
+- `Xc`: a TileDB array, of shape (n_var) for sparse data, with a single TileDB attribute of numeric type. The first ("") is expression data, and the second ("obs") are the corresponding row indices. Used for column-slicing.
+- `X_col_shift`: (optional) TileDB array used in column shift encoding, shape (n_var,), dtype = X.dtype. Single unnamed numeric attribute.
+- `emb`: a TileDB group, which in turn contains all (zero or more) embeddings.
+- `emb/<EMBEDDING_NAME>`: a TileDB array, with a single anonymous attribute, of numeric type, and shape (n_obs, N>=2).
+- `cxg_group_metadata`: an empty TileDB array, used to store CXG-wide metadata
+- `uns`: a TileDB group for `unstructured` dataset metadata. holidng Key->value pairs where key is a string and value is a serialized dictionary.
```

Comment:
New content here ☝🏽

Reply:
The value doesn't necessarily _need_ to be a serialized dictionary right? What does the backend currently support?

Reply:
Correct - any key-value pair